extensions/amp-a4a/0.1/legacy-signature-verifier.js

```
@@ -14,51 +14,302 @@
+                  TAG,
+                  `Invalid response from signing server ${signingServiceName}`);
+              return [];
+            }
+          }).then(jwks => ({
+            signingServiceName,
+            keys: jwks.map(jwk => this.importPublicKey_(signingServiceName, jwk)
+                .catch(err => {
+                  user().error(
+                      TAG, `error importing keys for: ${signingServiceName}`,
+                      err);
+                  return null;
+                })),
+          })).catch(err => {
+            user().error(TAG, err);
+            // TODO(a4a-team): This is a failure in the initial attempt to get
+            // the keys, probably b/c of a network condition.  We should
+            // re-trigger key fetching later.
+            return {signingServiceName, keys: []};
+          }));
+    } else {
+      // The given signingServiceName does not have a corresponding URL in
+      // _a4a-config.js.
+      const reason = `Signing service '${signingServiceName}' does not exist.`;
+      user().error(TAG, reason);
+      this.keys_.push(/** @type {!Promise<!CryptoKeysDef>} */ (Promise.resolve({
+        signingServiceName,
+        keys: [],
+      })));
+    }
+  }
+
+  /**
+   * Extracts a cryptographic signature from `headers` and verifies that it's
+   * the correct cryptographic signature for `creative`.
+   *
+   * As a precondition, `loadKeyset` must have already been called on the
+   * signing service that was used.
+   *
+   * @param {!ArrayBuffer} creative
+   * @param {!Headers} headers
+   * @param {function(string, !Object)} lifecycleCallback called for each AMP
+   *     lifecycle event triggered during verification
+   * @return {!Promise<?VerificationFailure>} resolves to `null` on success, or,
+   *     on failure, to a `VerificationFailure` indicating the cause
+   */
+  verify(creative, headers, lifecycleCallback) {
+    if (!this.isAvailable_()) {
+      return Promise.resolve(
```

I was referring to https://github.com/ampproject/amphtml/pull/10674/files#diff-b139e3c4641f6d160f2832cba8be0761R666, where for KEY_NOT_FOUND and SIGNATURE_MISMATCH you do:
```
  user().error(
       TAG, this.element.getAttribute('type'),
       'Signature verification failed');
```
...but for NO_FAULT, there is no user().error(...). My concern is that we don't want a silent failure if there is a network issue. So maybe in the NO_FAULT case of that switch include the same error message. Or you could log "Signature verification failed because signature verifier is unavailable" either in the switch or here in verify().